### PR TITLE
update the translation of `Array.toString()`

### PR DESCRIPTION
**Pull request by robert-jx**
### Description

update the translation of array.toString()
### Motivation

update the translation of array.toString()
### Additional details

none

### CODE REVIEW

**Comment by skyclouds2001** [comment]
I think "See also" should be translated to "参见" would to better
[See the guide document](https://github.com/mdn/translated-content/blob/main/docs/zh-cn/translation-guide.md#%E5%B8%B8%E7%94%A8%E6%A0%87%E9%A2%98)
@robert-jx

**Comment by robert-jx** [comment]
@JasonLamv-t

File: files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md [hunk 1]
```diff
@@ -17,31 +17,51 @@ toString()
 
 ### 返回值
 
-一个表示数组所有元素的字符串。
+一个表示数组元素的字符串。
 
 ## 描述
 
-{{jsxref("Array")}} 对象覆盖了 {{jsxref("Object")}} 的 `toString` 方法。对于数组对象，`toString` 方法在内部调用 [`join()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/join) 方法拼接数组中的元素并返回一个字符串，其中包含用逗号分隔的每个数组元素。如果 `join` 方法不可用，或者它不是一个函数，将使用 [`Object.prototype.toString`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/toString) 代替，返回 `[object Array]`。
+{{jsxref("Array")}} 对象会覆盖 {{jsxref("Object")}} 的 `toString` 方法。数组的 `toString` 方法在内部调用 [`join()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/join)，它连接数组并返回一个字符串，其中包含以逗号分隔的每个数组元素。如果 `join` 方法不可用或者不是函数，将会使用 [`Object.prototype.toString`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/toString) 来代替, 并返回 `[object Array]`。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md:24]:
```suggestion
{{jsxref("Array")}} 对象覆盖了 {{jsxref("Object")}} 的 `toString` 方法。数组的 `toString` 方法实际上在内部调用了 [`join()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/join) 方法来拼接数组并返回一个包含所有数组元素的字符串，元素之间用逗号分隔。如果 `join` 方法不可用或者不是函数，则会使用 [`Object.prototype.toString`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/toString) 来代替, 并返回 `[object Array]`。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md [hunk 2]
```diff
@@ -17,31 +17,51 @@ toString()
...
 
 ```js
 const arr = [];
-arr.join = 1; // re-assign `join` with a non-function
-console.log(arr.toString()); // Logs [object Array]
+arr.join = 1; // 用非函数重新分配 `join`。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md:28]:
```suggestion
arr.join = 1; // 将 `join` 重新赋值为非函数的值。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md [hunk 3]
```diff
@@ -17,31 +17,51 @@ toString()
...
+console.log(arr.toString()); // [object Array]
 
-console.log(Array.prototype.toString.call({ join: () => 1 }));  // Logs 1
+console.log(Array.prototype.toString.call({ join: () => 1 })); // 1
 ```
 
-当一个数组被作为文本值或者进行字符串拼接操作时，将会自动调用其 `toString` 方法。
+当一个数组被表示为文本值或一个数组被引用为字符串连接时，JavaScript会自动调用 `toString` 方法。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md:34]:
```suggestion
当数组需要被表示为文本值，或者当数组在字符串拼接中被引用时，JavaScript 会自动调用 toString() 方法。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md [hunk 4]
```diff
@@ -17,31 +17,51 @@ toString()
...
+const array1 = [1, 2, "a", "1a"];
 
-console.log(array1.toString());
-// expected output: "1,2,a,1a"
+console.log(array1.toString()); // "1,2,a,1a"
+```
+
+### 在稀疏数组中使用 toString()
+
+遵循 `join()` 的行为，`toString()` 将空槽视为 `undefined` 并生成额外的分隔符：
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md:48]:
```suggestion
与 `join()` 的行为一致，`toString()` 将空槽视为 `undefined` 并生成一个额外的分隔符：
```

File: files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md [hunk 5]
```diff
@@ -17,31 +17,51 @@ toString()
...
+
+```js
+console.log([1, , 3].toString()); // '1,,3'
+```
+
+### 在非数组对象中使用 toString()
+
+`toString()` 是 [通用的](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#通用数组方法)。它期望 `this` 具有 `join()` 方法；否则，使用 `Object.prototype.toString()` 来代替。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md:56]:
```suggestion
`toString()` 是[通用的](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#通用数组方法)。它期望 `this` 具有 `join()` 方法；如果不存在，则使用 `Object.prototype.toString()`。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md [hunk 6]
```diff
@@ -17,31 +17,51 @@ toString()
...
+
+```js
+console.log(Array.prototype.toString.call({ join: () => 1 }));
+// 1; a number
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/tostring/index.md:60]:
```suggestion
// 1; 一个数字
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions

**Review by yin1999 — APPROVED** [review]

**Pull request merged**